pkg/kubelet/nodestatus/setters.go

```diff
@@ -415,6 +415,10 @@ func Images(nodeStatusMaxImages int32,
 			if len(names) > MaxNamesPerImageInNodeStatus {
 				names = names[0:MaxNamesPerImageInNodeStatus]
 			}
+			// Use the image ID as fallback if no repo tag or digest exists
+			if len(names) == 0 {
+				names = []string{image.ID}
```

Comment:
while I'm in favor of adding the image id, it seems like we're conflating IDs and name in this case, maybe we should have a field for imageID

Reply:
Yeah why not, that sounds reasonable to me too. How to get a decision about that?

Reply:
cc @yujuhong @dashpole

Reply:
Agree with Yassine about the possible danger of conflating image IDs with names. Just a thought, but could the `Names` field in the `v1.ContainerImage` struct be simply marked as `omitempty`?

https://github.com/kubernetes/kubernetes/blob/518f16636f08046ae1dca84d5fdb016c70ad37ef/staging/src/k8s.io/api/core/v1/types.go#L4148

I mean, just like the `Node.Status.Images` field is annotated as `omitempty`:

https://github.com/kubernetes/kubernetes/blob/518f16636f08046ae1dca84d5fdb016c70ad37ef/staging/src/k8s.io/api/core/v1/types.go#L4088

Reply:
Yeah I thought this too but then we would only have a size for images, which does not bring any information to the customer in my point of view. Afterwards I thought about displaying the ID, because that is a unique identifier to trace on the host if needed.

Reply:
ack. yeah, a new optional ID field on ContainerImage is probably the best option. I don't know how (if?) a v1 API resource model can be changed, though :(

Reply:
Do I have to write a KEP for this? Changing the API was something I'd like to avoid in the first place to fix this issue. What about skipping the images for now?

Reply:
@jaypipes - as long as the change is backward compatible (more info: https://kubernetes.io/docs/reference/using-api/deprecation-policy/)

Reply:
I agree we should not put data with a different meaning in an existing field.

To take a step back, including image names in node status was probably not a great idea to begin with (it *significantly* increased node API object size, which led to performance and etcd issues, which meant we had to truncate the list, which means the data is not complete, which makes it unfit for many uses). Before we expand API surface around this, we should verify with sig-node/sig-scalability that providing this data is desired long-term

Reply:
And yes, API changes require KEPs (https://github.com/kubernetes/community/blob/master/sig-architecture/api-review-process.md#mechanics)

Reply:
> Do I have to write a KEP for this? Changing the API was something I'd like to avoid in the first place to fix this issue. What about skipping the images for now?

You mean skipping images that do not have a name? Not sure... I find it odd actually that an image would have no name, but I suppose the situation does indeed exist.

By skipping images that do not have a name, you will basically hide information from the end user (images are on the node but the user would not know that). Not sure if that's a worse experience than the bug here (of the Node.Status structure not being proper)